iothub/device/src/ClientProperties.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// A container for properties retrieved from the service.
+    /// <remarks>
+    /// The Properties class is not meant to be constructed by customer code.
+    /// It is intended to be returned fully populated from the client method.
+    /// </remarks>
+    /// </summary>
+    public class ClientProperties : ClientPropertyCollection
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ClientProperties"/>.
+        /// </summary>
+        internal ClientProperties()
```

Comment:
If this is public, you should add a public constructor that does nothing (and say for test purposes only) if there is value in this class being mockable.

Reply:
no harm in making this mockable anyway, I'll add a public no args constructor